job-server/test/spark.jobserver/JobManagerSpec.scala

```diff
@@ -195,5 +196,20 @@ abstract class JobManagerSpec extends JobSpecBase(JobManagerSpec.getNewSystem) {
         case JobResult(_, result: Int) => result should equal (1 + 2 + 3)
       }
     }
+
+    it("should be able to cancel running job") {
```

Comment:
@zeitos It seems from the code that deleting a nonexistant job would get the user the NotFound message.   I think the same thing if the job already finished.